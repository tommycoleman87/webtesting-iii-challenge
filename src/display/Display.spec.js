// Test away!
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Display from './Display';

describe('<Display />', () => {
    it('matches snapshot', () => {
        const tree = renderer.create(<Display />);

        expect(tree.toJSON()).toMatchSnapshot();
    })

    it('diplay closed if the closed prop is true or open if the closed prop is false', () => {
        const status = {
            closed: true,
        }
        const display = render(<Display closed={status.closed}/>)
        const open = display.getByTestId('isOpen');
        expect(open.innerHTML).toEqual('Closed')
    })

    it('displays locked it the locked prop is true or unlocked if the locked prop is false ', () => {
        const status = {
            locked: true,
        }

        const display = render(<Display closed={status.closed} locked={status.locked}/>)

        const locked = display.getByTestId('isLocked');
        expect(locked.innerHTML).toEqual('Locked')
    })

    it('should have red-led class when locked or closed', () => {
        
    })

    it('should have green-led class when unlocked or open', () => {
        
    })
})