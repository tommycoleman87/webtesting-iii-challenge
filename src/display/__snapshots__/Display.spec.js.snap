// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="isLocked"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="isOpen"
  >
    Open
  </div>
</div>
`;
